File: examples/mpi-hello.f90
program hello_world_mpi
include 'mpif.h'

integer :: nprocs, myid, ierr      
! Initializing the MPI environment
call MPI_INIT(ierr)

! The total size of environment in terms of process quantity
call MPI_COMM_SIZE(MPI_COMM_WORLD, nprocs, ierror)

! Returns process ID for each process
call MPI_COMM_RANK(MPI_COMM_WORLD, myid, ierror)

! Printing Hello World from each process & reporting its rank
print*, 'Hello, Fortran World! from process: ', myid, 'of ', nprocs 

! Finalizing the MPI environment
call MPI_FINALIZE(ierr)

end program

File: examples/mpi-hello.c
#include <mpi.h>
#include <stdio.h>

int main(int argc, char** argv) {
    //  Initialize MPI environment
    MPI_Init(NULL, NULL);

    // Decide the number of processes
    int nprocs;
    MPI_Comm_size(MPI_COMM_WORLD, &nprocs);
    
    // Rank of the process
    int myid;
    MPI_Comm_rank(MPI_COMM_WORLD, &myid);


    // printf() displays the string inside quatation
    printf("Hello, C World! from processor %d out of %d processors\n",
           myid, nprocs);

    // Finalize MPI environment
    MPI_Finalize();

    return 0;
}

